### PR TITLE
chore: auto-publish package to npm after release

### PR DESCRIPTION
**Pull request by targos**

### CODE REVIEW

**Comment by targos** [comment]
Labeled "do not land" because we need to setup the npm token if this is accepted.

**Comment by mmarchini** [comment]
This would remove 2FA when we publish. Are we ok with that?

**Comment by targos** [comment]
npm now has automation tokens now, so we don't have to disable 2FA requirements on the package. AFAIU the only risk we would have is that someone with write access to this repo pushes something to steal the token?

**Comment by mmarchini** [comment]
We still drop 2FA for our "main" publish workflow (assuming the automation becomes the main workflow). It's not ideal IMO but probably low risk enough that we can try it?

**Comment by targos** [comment]
GitHub Actions now have environment protection rules and environment secrets: https://github.blog/changelog/2020-12-15-github-actions-environments-environment-protection-rules-and-environment-secrets-beta/
Maybe that can be used to protect an npm token and/or only allow a subset of collaborators to publish the package?

File: .github/workflows/release-please.yml
Comment (Trott) [inline .github/workflows/release-please.yml:23]:
```suggestion
      - uses: actions/setup-node@v3
        with:
          node-version: 16.x
```

**Comment by targos** [comment]
I will reimplement the change differently if it's likely to land.

**Comment by targos** [comment]
Updated. We still need to setup an npm automation token before merging this. Who would be able to do that?

**Comment by targos** [comment]
The NPM_TOKEN secret is installed. Would you like to review again?

File: .github/workflows/release-please.yml
Comment (targos) [inline .github/workflows/release-please.yml:32]:
```suggestion
          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

**Review by richardlau — APPROVED** [review]

**Pull request merged**
